kindle.py

```diff
@@ -192,26 +200,30 @@ def download_one_book(self, asin, device, index, filetype="EBOK"):
             total_size = r.headers["Content-length"]
             out = os.path.join(self.out_dir, name)
             logger.info(
-                f"({index}/{self.total_to_download})downloading {name} {total_size} bytes"
-            )
+                    f"({index}/{self.total_to_download})downloading {name} {total_size} bytes")
             with open(out, "wb") as f:
                 for chunk in r.iter_content(chunk_size=512):
                     f.write(chunk)
             logger.info(f"{name} downloaded")
         except Exception as e:
-            logger.info(str(e))
-            logger.info(f"{asin} download failed")
+            if tries > 0:
+                logger.warn(f"{asin} [{name}] download failed, try again.")
+                time.sleep(5)
+                self.download_one_book(asin, device, index, tries - 1, filetype=filetype)
+            else:
+                logger.error(f"{asin} [{name}] download failed, skipping.")
 
-    def download_books(self, start_index=0, filetype="EBOK"):
+    def download_books(self, start_index=0, filetype="EBOK", max_tries=5):
         # use default device
         device = self.get_devices()[0]
         asins = [book["asin"] for book in self.get_all_books(filetype=filetype)]
         self.total_to_download = len(asins) - 1
         if start_index > 0:
-            print(f"resuming the download {start_index}/{self.total_to_download}")
+            logger.info(f"resuming the download {start_index}/{self.total_to_download}")
```

Comment:
I prefer `print` here what do you think?

Reply:
I changed the print to `logger` so it can be easier to redirect (for example, in GUI)

Reply:
agree